src/rune-bridge/src/core/index_provider.rs

```diff
@@ -94,78 +103,24 @@ struct RunesResponse {
 }
 
 /// Implementation of the `RuneIndexProvider` trait that uses the `HttpClient` to make requests to
-pub struct OrdIndexProvider<C: HttpClient> {
+pub struct OrdIndexProvider<C: HttpClient + Sync> {
     client: C,
-    indexer_urls: HashSet<String>,
-    /// Minimum quantity of indexer nodes required to reach agreement on a
-    /// request
-    pub indexer_consensus_threshold: u8,
+    url: String,
 }
 
 impl<C> OrdIndexProvider<C>
 where
-    C: HttpClient,
+    C: HttpClient + Sync,
 {
-    pub fn new(client: C, indexer_urls: HashSet<String>, indexer_consensus_threshold: u8) -> Self {
-        Self {
-            client,
-            indexer_urls,
-            indexer_consensus_threshold,
-        }
-    }
-
-    /// Get consensus response from the indexer.
-    ///
-    /// All indexers must return the same response for the same input, other
-    /// the function will return an error.
-    async fn get_consensus_response<T>(&self, uri: &str) -> Result<T, GetInputsError>
-    where
-        T: Clone + DeserializeOwned + PartialEq + Debug,
-    {
-        let mut failed_urls = Vec::with_capacity(self.indexer_urls.len());
-        let mut responses: Vec<(String, T)> = Vec::new();
-        let mut indexers_agree = true;
-
-        for url in &self.indexer_urls {
-            match self.client.http_request::<T>(url, uri).await {
-                Ok(response) => {
-                    if !responses.is_empty() && responses[0].1 != response {
-                        indexers_agree = false;
-                    }
-
-                    responses.push((url.clone(), response));
-                }
-                Err(e) => {
-                    log::warn!("Failed to get response from indexer {}: {:?}", url, e);
-                    failed_urls.push(url.clone());
-                }
-            }
-        }
-
-        if responses.len() < self.indexer_consensus_threshold as usize {
-            Err(GetInputsError::InsufficientConsensus {
-                received_responses: responses.len(),
-                required_responses: self.indexer_consensus_threshold,
-                checked_indexers: self.indexer_urls.len(),
-            })
-        } else if !indexers_agree {
-            // TODO: After https://infinityswap.atlassian.net/browse/EPROD-971 is done, return
-            // actual values here instead of formated response
-            Err(GetInputsError::IndexersDisagree {
-                indexer_responses: responses
-                    .into_iter()
-                    .map(|(url, response)| (url, format!("{response:?}")))
-                    .collect(),
-            })
-        } else {
-            Ok(responses.pop().expect("responses vector is empty").1)
-        }
+    pub fn new(client: C, url: String) -> Self {
+        Self { client, url }
     }
 }
 
+#[async_trait]
 impl<C> RuneIndexProvider for OrdIndexProvider<C>
 where
-    C: HttpClient,
+    C: HttpClient + Sync + Send,
```

Comment:
Is it really require client sharing? Maybe better to make it `#[async_trait(?Send)]`?